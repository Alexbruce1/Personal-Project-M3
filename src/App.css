body {
  background-color: #666;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.parallax-header {
  left: 0;
  margin: 0;
  top: 0;
  width: 100vw;
}

.parallax-image {
  height: 100vh;
  /* position: absolute; */
  /* top: 0; */
  /* left: 0; */
  margin: 0;
  padding: 0;
  width: 100vw;
  z-index: -1;
}

.parallax-image img {
  /* filter: blur(3px); */
  height: 150%;
  position: relative;
  right: 1px;
  top: 40%;
  width: calc(100% + 1px);
}