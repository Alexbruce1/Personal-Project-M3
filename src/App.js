import React, { Component } from 'react';
import './App.css';
import backArrow from './images/back.svg'
import CardContainer from './CardContainer/CardContainer';
import Header from './Header/Header';
import Search from './Search/Search';
import { Route, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { searchData } from './utilities/Cleaner';
import { Parallax, Background } from 'react-parallax';
import { setData, removeStateMedia } from './actions/actions';


export class App extends Component {
  constructor(props) {
    super();
    this.state = {
      // data: [],
      query: '',
      mediaType: '',
    }
  }

  storeQuery = e => {
    this.setState({
      query: e.target.value
    })
  }

  setType = e => {
    this.setState({
      mediaType: e.target.value
    })
  }

  callApi = async() => {
    const { query, mediaType } = this.state;
    const returnedData = await searchData(query, mediaType);
    this.props.setData(returnedData);
  }

  removeMedia = () => {
    let newMedia = []
    this.props.removeStateMedia(newMedia)
  }

  render() {
    const { media } = this.props;
    return (
      <div className='App'>
        <Route exact path='/' render={() => {
          return(
            <Parallax
              strength={700}
              className='parallax-header'
            >
              <Background className='parallax-image'>
                <div className='parallax-transition'></div>
              </Background>
              <Header />
            <NavLink className='navlink-search' to='/search' >Search for Content</NavLink>
          </Parallax>
          )
        }} />
        <Route exact path='/search' render={() => {
          return (
            <div className='App'>
              <Parallax
                strength={700}
                className='parallax-header'
              >
                <Background className='parallax-image'>
                  <div className='parallax-transition'></div>
                </Background>

                <Header />
                {
                  !media.length &&
                  <Search
                    callApi={this.callApi}
                    storeQuery={this.storeQuery}
                    setType={this.setType}
                  />
                }
              </Parallax>
            </div>
          )
        }} />
        {
          media.length &&
          <Route path='/search' render={() => {
            return (
              <div>
                <button className='remove-media' onClick={() => { this.removeMedia() }}>
                  <img alt='' src={backArrow} />
                  Make Another Search
                </button>
                <CardContainer media={media} />
              </div>
            )
          }} />
        }
      </div>
    );
  }
}

export const mapStateToProps = ({ media }) => ({
  media
});

export const mapDispatchToProps = (dispatch) => ({
  setData: (media) => dispatch(setData(media)),
  removeStateMedia: (newMedia) => dispatch(removeStateMedia(newMedia)),
});


export default connect( mapStateToProps, mapDispatchToProps )(App);