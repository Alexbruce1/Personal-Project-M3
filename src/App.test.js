import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let data;

  beforeEach(() => {
    data = []
    wrapper = shallow(<App media={data} />);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show media', () => {
    const initialState = {
      media: []
    };

    expect(mapStateToProps(initialState).media).toEqual([]);
  });

  it('should set data into redux state', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).setData();

    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SET_DATA' });
  });

  it('should remove data from redux state', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).removeStateMedia();

    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'REMOVE_STATE_MEDIA' });
  });
  
  // it('renders without crashing', () => {
  //   // const div = document.createElement('div');
  //   // ReactDOM.render(<App />, div);
  //   // ReactDOM.unmountComponentAtNode(div);
  // });
  
})