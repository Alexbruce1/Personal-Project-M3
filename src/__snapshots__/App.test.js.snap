// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    media={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/search"
          render={[Function]}
        />,
        0,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/search",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      0,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/search"
            render={[Function]}
          />,
          0,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/search",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        0,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
